base/missing.jl

```diff
@@ -274,63 +274,72 @@ mapreduce(f, op, itr::SkipMissing{<:AbstractArray}) =
 
 function _mapreduce(f, op, ::IndexLinear, itr::SkipMissing{<:AbstractArray})
     A = itr.x
-    local ai
+    ai = missing
     inds = LinearIndices(A)
     i = first(inds)
     ilast = last(inds)
-    while i <= ilast
+    for outer i in i:ilast
         @inbounds ai = A[i]
-        ai === missing || break
-        i += 1
+        ai !== missing && break
     end
-    i > ilast && return mapreduce_empty(f, op, eltype(itr))
+    ai === missing && return mapreduce_empty(f, op, eltype(itr))
     a1::eltype(itr) = ai
+    i == typemax(typeof(i)) && return mapreduce_first(f, op, a1)
```

Comment:
Could this even be changed to this?
```suggestion
    i == ilast && return mapreduce_first(f, op, a1)
```

Reply:
Yes, sure. Everything works under the assumption that `ifirst <= ilast`, so if `i != ilast` then it must be `i < ilast <= typemax(typeof(i))` and there is no overflow in the subsequent `i += 1`.